TЗ - 10.

shepilovaElena opened the pull request:
version 1.

shepilovaElena committed:
TЗ - 10.
version 1.

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,61 @@
+    private Map<Integer, Film> films = new HashMap<>();
+
+    private int getNextId() {
+        if (counter == 0) {
```
так может просто counter начать сразу с 1 и не мучиться с условиями?))

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,61 @@
+    @PostMapping
+    public Film postFilm(@Valid @RequestBody Film film) throws ValidationException {
+        log.info("Получен запрос на добавление фильма.");
+        if (!(film.getReleaseDate() == null) && film.getReleaseDate().isBefore(LocalDate.of(1895, 12, 28))) {
```
условие !(film.getReleaseDate() == null) лучше заменить на легко читаемое film.getReleaseDate() != null 
+ 
эта валидация нужна и при обновлении фильма, поэтому лучше вынести во внутренний метод, чтобы не дублировать код

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,64 @@
+    private int counter;
+
+    private int getNextId() {
+        if (counter == 0) {
```
в это классе тоже можно counter сразу начать с 1

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,64 @@
+    @PostMapping
+    public User postUser(@Valid @RequestBody User user) {
+        log.info("Получен запрос на добавление пользователя.");
+        if (user.getLogin().contains(" ")) {
```
эта валидация нужна и при обновлении пользователя, поэтому лучше вынести во внутренний метод, чтобы не дублировать код

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,64 @@
+            log.warn("Логин пользователя {} содержит пробел.", user.getLogin());
+            throw new ValidationException("Логин не может содержат знак пробела.");
+        }
+        if (user.getName() == null) {
```
и эта логика тоже

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/ValidationException.java:
```diff
@@ -0,0 +1,10 @@
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
```
в этом классе логирование не нужно

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,12 +1,29 @@
-@Setter
+@Data
+@Builder
+@Slf4j
```
как и тут

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,12 +1,29 @@
+    @NotBlank(message = "Описание не должно быть пустым.")
+    @Size(message = "Описание не должно превышать 200 символов.", max = 200)
+    private String description;
+    private LocalDate releaseDate;
```
для этого поля нужна аннотация @NotNull с комментом

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,25 @@
+
+@Data
+@Builder
+@Slf4j
```
в этом классе логирование не нужно

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,25 @@
+public class User {
+    private int id = 0;
+    @Email(message = "email введен некорректно.")
+    private String email;
```
для этого поля нужна аннотация @NotNull с комментом

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,25 @@
+    private String login;
+    private String name;
+    @Past(message = "Дата рождения не может быть в будущем.")
+    private LocalDate birthday;
```
и для этого тоже

shepilovaElena committed:
TЗ - 10.
класс  FilmController и класс UserController - исправлена логика счетчик id (метод getNextId()).

класс  FilmController - добавлен метод checkFilmReleaseDate().

класс UserController - добавлен метод checkUserLogin().

Исправление тестов.

shepilovaElena committed:
TЗ - 10.
